src/main/java/canelo/Canelo.java

```diff
@@ -0,0 +1,314 @@
+
+    static final String BY = "/by";
+    static final String FROM = "/from";
+    static final String TO = "/to";
+
+    static final int MINIMUM_TODO_LENGTH = 5;
+    static final int MINIMUM_MARK_LENGTH = 5;
+    static final int MINIMUM_UNMARK_LENGTH = 7;
+    static final int MINIMUM_DEADLINE_LENGTH = 9;
+    static final int DEADLINE_BY_LENGTH = 4;
+    static final int MINIMUM_EVENT_LENGTH = 6;
+    static final int EVENT_TO_LENGTH = 4;
+    static final int MINIMUM_DELETE_LENGTH = 7;
+
+    static final String LINE = "____________________________________________________________";
+
+    private static void printStartMessage() {
+        System.out.println(LINE);
+        System.out.println("""
+                 Hello! I'm Canelo :)
+                 What can I do for you?""");
+        System.out.println(LINE);
+    }
+
+    private static void printEndMessage() {
+        System.out.println(LINE);
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(LINE);
+    }
+
+    private static boolean isValidUserInput(String userInput) {
+        return !userInput.equals("bye") && !userInput.equals("q");
+    }
+
+    private static String determineTaskType(String userInput) {
+        String[] splitKeyboardInput = userInput.split(" ");
+        return switch (splitKeyboardInput[0]) {
+            case "list" -> "list";
+            case "mark" -> "mark";
+            case "unmark" -> "unmark";
+            case "todo" -> "todo";
+            case "deadline" -> "deadline";
+            case "event" -> "event";
+            case "delete" -> "delete";
+            default -> "unknown";
+        };
+    }
+
+    private static void handleList() {
+        for (int i = 0; i < numTasks; i++) {
+            int taskNumber = i + 1;
+            String typeIcon = tasks.get(i).getTypeIcon();
+            String statusIcon = tasks.get(i).getStatusIcon();
+            String description = tasks.get(i).getDescription();
+            System.out.println(taskNumber + ".[" + typeIcon + "][" + statusIcon + "] " + description);
+        }
+    }
+
+    private static void handleMark(String userInput) throws CaneloException {
+        if (userInput.length() <= MINIMUM_MARK_LENGTH) {
+            throw new CaneloException("Please add a task number to mark.");
+        }
+        int taskNumber = Integer.parseInt(userInput.substring(MINIMUM_MARK_LENGTH));
+        if (isInvalidTaskNumber(taskNumber)) {
+            throw new CaneloException("Please input a valid task number between 1 and " + numTasks + ".");
+        }
+        Task task = tasks.get(taskNumber - 1);
+        task.markDone();
+        System.out.println("Nice! I've marked this task as done:\n[" + task.getTypeIcon() + "][X] " + task.getDescription());
+        try {
+            writeToCaneloFile();
+        } catch (IOException e) {
+            System.out.println("Something went wrong when writing to Canelo File: " + e.getMessage());
```

Comment:
there are many things being done in this method. consider extracting them out to improve the SLAP